src/cmd/ksh93/tests/b_sync.sh

```diff
@@ -23,9 +23,9 @@ expect=""
 #
 # We don't use fd 3 because it might be opened by a debug malloc subsystem. We expect fd 999 to be
 # unused if not invalid.
-actual=$(sync -s 999 2>&1)
+actual=$(sync -s999 2>&1)
```

Comment:
If the removal of the space between the short option and its value changes the behavior that means `getopt_long()`, or its use by `b_sync()`, is seriously broken. That is unlikely given that it works correctly on quite a few platforms. What is more likely is there is something odd about your test environment. Perhaps your virtual-machine or other sandbox behavior that makes `fsync(999)` seem to succeed when it should fail.

P.S., the `log_error()` portion of your change is correct; albeit solely to avoid confusion for a human reader of the test.

Reply:
I agree the space between the -s and 999 doesn't have any impact, just thought I'd make it consistent with the prior test. I only wanted to update the log message as it confused me initially while reviewing the output.

Reply:
It is actually a net positive that tests vary such seemingly trivial details as the spacing between options and their value. Attempting 100% consistency on such points, that are otherwise irrelevant to the real purpose of the test, is counter-productive. Such inconsistencies help to find bugs and help ensure readers of the code do not make unwarranted assumptions.

Reply:
It has been reverted...